client/orderbook/orderbook.go

```diff
@@ -220,20 +220,53 @@ func (ob *OrderBook) processCachedNotes() error {
 		}
 	}
 
+	// Must be done with ob.noteQueueMtx locked, otherwise some notes might be sent
+	// to ob.noteQueue after we are done processing it. Such notes won't be processed
+	// until next processCachedNotes call when they'll be irrelevant and more
+	// importantly it means we won't apply them on top of server snapshot they were
+	// meant to be applied to.
+	ob.setSynced(true)
+
 	return nil
 }
```

Comment:
Since cache isn't used (based on what I observed in the logs, and my current understanding of the relevant code), this fix is non-effectual really. 

Moreover, we might not even need caching here really, so in the next PR I'll consider getting rid of it.

Reply:
Thoughts on whether or not we need (to fix or reimplement) that caching there:
- there is a 1024+128 buffer in `comms` package to hold incoming WS messages, it seems like enough of a buffer to hold those server update notifications off for a while (and probably we can easily increase these number if necessary, probably want to add some `Tracef` logs to monitor saturation in case it gets dangerously high during network delays and whatnot)
- ALL server notifications will be waiting behind one that got blocked (waiting on `dc.booksMtx` for example), it doesn't seems to be a big deal here cause we don't have to handle them in real time